Controllers/StudentsController.cs

```diff
@@ -28,5 +31,32 @@ public StudentsController(IStudentRepo studentService)
             // instead of 200(Ok)
             // return CreatedAtRoute(nameof(GetStudentById), new {Id = student.Id}, student);
         }
+
+        //GET api/students
+        [HttpGet]
+        public ActionResult <IEnumerable<Student>> GetStudents(){
+            var students = _studentService.GetAllStudent();
+            return Ok(students);
+        }
+
+        //GET api/students/{id}
+        [HttpGet("{id}", Name = "GetStudentById")]
+        public ActionResult <Student> GetStudentById(int id){
+            var student = _studentService.GetStudentById(id);
+            if(student != null){
+                return Ok(student);
+            }
+            return NotFound();
+        }
+
+        //GET api/students/{keyword}
+        [HttpGet("Search")]
```

Comment:
should search be a GET or POST?
read about it and give me an informed decision

Reply:
GET is the appropriate verb for search operation. The use of GET is to list collection or retrieve resource/item. Since the parameter is on the query part of the URL it makes it user friendly and ease the sharing of search URL.

Reply:
is it RESTFUL?